### ISSUE TITLE
[ENH] Add dedicated benchmarking module for time series algorithms

### ISSUE THREAD

**Issue by OmBiradar** [sktime/sktime#8103]
**Is your feature request related to a problem? Please describe.**

Currently, sktime lacks a standardized and comprehensive way to benchmark different time series algorithms across datasets. While there is existing functionality in Benchmarking, users typically have to write custom scripts to compare models systematically. This leads to:

- Inconsistent evaluation methodologies
- Difficulty reproducing benchmark results
- Challenges in model selection for new users
- Lack of standard baselines for algorithm developers

This gap makes it harder for practitioners to make informed decisions about which algorithms to use for their specific time series problems.

**Describe the solution you'd like**

I propose adding a dedicated benchmarking module inspired by scikit-learn's benchmark directory but tailored for time series tasks. The implementation would:

1. Create a set of benchmark scripts following a consistent structure:

- bench_forecasting_naive.py
- bench_forecasting_arima.py
- bench_forecasting_exp_smoothing.py
- bench_tsc_distance_based.py
- bench_ucr_archive.py (dataset-specific)

2. Each script would follow a consistent pattern:

- Command-line interface for selecting models/datasets
- Proper time series cross-validation strategies
- Standard metrics relevant to the task
- Performance timing (fit/predict times)
- Formatted tabular output

3. Integration with existing sktime components:

- Leverage ForecastingBenchmark class
- Use standard sktime CV splitters
- Utilize sktime metrics
- Compatible with all sktime estimators

**Describe alternatives you've considered**

1. Jupyter notebooks: While flexible, they lack standardization and are less reproducible from the command line.

2. Web-based dashboards: These would be harder to maintain and integrate into workflows.

**Additional context**

This proposal is inspired by scikit-learn's benchmarking directory structure, which has proven valuable for algorithm comparison and development. Having standardized benchmarks would:

- Help new users select appropriate algorithms
- Provide baseline performance expectations
- Support algorithm developers in measuring improvements
- Facilitate reproducible research

Example usage would be:

```bash
python bench_forecasting_naive.py -s last mean drift -d airline shampoo
```

This would complement sktime's existing educational resources and make time series analysis more accessible to practitioners at all levels.

I am a newbie so you can understand if there is some mistake in my thoughts.

Please give your feedback!

**Comment by fkiraly** [comment]
FYI @benHeid, @jgyasu, @felipeangelimvieira

**Comment by jgyasu** [comment]
> FYI [@benHeid](https://github.com/benHeid), [@jgyasu](https://github.com/jgyasu), [@felipeangelimvieira](https://github.com/felipeangelimvieira)

From what I understand, isn't this more focused on the end goal of the benchmarking framework, i.e., having a leaderboard of sort that we discussed the other day? Please correct me if I am wrong.

**Comment by OmBiradar** [comment]
@fkiraly, thank you for your feedback and questions.

Yes, I'm referring to reproducible benchmarks that produce results using the orchestrator, not just the orchestrator itself. The goal I have is to have standardized scripts that can be run to compare different algorithms in a consistent way.

Regarding scikit-learn's benchmark directory, it's located [here](https://github.com/scikit-learn/scikit-learn/tree/main/benchmarks)

It contains various benchmark scripts organized by algorithm type and task, which provide a standardized way to measure and compare performance.

> FYI [@benHeid](https://github.com/benHeid), [@jgyasu](https://github.com/jgyasu), [@felipeangelimvieira](https://github.com/felipeangelimvieira)

**Comment by OmBiradar** [comment]
Hey @jgyasu , Yes! It's kinda more focused on the end goal of benchmarking i.e. having a leaderboard. Personally I think this would be a step before having a leaderboard. I think this can serve as the infrastructure that could feed into the leaderboard system. These scripts would provide the code required to populate the leaderboard.

**Comment by fkiraly** [comment]
Interesting idea, @OmBiradar.

As far as I understand, the benchmarks in `sklearn` seem to be primarily used to track speed changes across versions, they use airspeed:
https://scikit-learn.org/scikit-learn-benchmarks/

This is for performance reasons primarily, if I understand.
Would be nice to have - but a major project.

It might even be extremely interesting to look at this from an abstract API perspective, and see how it can be done in a more programmatic way across packages!

There is also another use case which is task performance rather than runtime - that would be closer to the leaderboard idea that @jgyasu is referring to. Is there anything like that in sklearn, @OmBiradar?

**Comment by OmBiradar** [comment]
@fkiraly, I think all of the ideas are great!

Firstly about `sklearn`:

I found that they have standalone benchmarking scripts along with benchmarking suit made to work with ASV.

Specifically, sklearn has two distinct benchmarking approaches:
- The `benchmarks/` directory contains standalone scripts for manual, ad-hoc benchmarking that generate immediate results with visualizations
- The `asv_benchmarks/` directory contains the official suite using Airspeed Velocity (ASV) for continuous performance monitoring across commits

Now I think sktime could have the following:

- The standalone suit compares models in a particular commit, allowing users to directly benchmark different algorithms on various datasets
- The ASV benchmark compares models across commits, helping track performance improvements or regressions over time

As both seem pretty interesting and important in their own ways...

For the standalone benchmarks, we could create scripts that:
- Compare different forecasting/classification algorithms on standard datasets
- Generate comprehensive metrics (accuracy, runtime, etc.)
- Produce visualizations for easy comparison
- Follow consistent patterns for reproducibility

For the ASV benchmarks, we could:
- Set up continuous monitoring of key algorithms
- Track performance across versions
- Integrate with CI/CD to detect regressions

The leaderboard could be implemented using the generated benchmarking results, similar to what GIFT-Eval does for time series forecasting. This would allow users to easily see which models perform best on specific datasets or tasks.

An additional benefit is that systematic benchmarking would help identify bugs or edge cases in new models. When a model suddenly performs much worse on standard benchmarks, it's often a sign of a regression bug that might not be caught by regular unit tests.

**Comment by OmBiradar** [comment]
Could I proceed to implement it?

**Comment by fkiraly** [comment]
> Could I proceed to implement it?

Sure, though I think this needs a good plan - it is a big projects. We should try to break it down in smaller steps, and we also want to create a design document first.

SInce it is not exactly clear how it would look like, code-wise.

Here is the repository where we curate design documents: https://github.com/sktime/enhancement-proposals/

**Comment by OmBiradar** [comment]
Yes, sure! Thanks for the info! I'll start making the **STEP**
